### DISCUSSION TITLE
Calculate next schedule

### DISCUSSION THREAD

**Discussion by mikeysan** [EddieHubCommunity/support#2169] [Ideas]
I would like some help with this idea. A colleague of mine and I wanted to figure out if we would be free to attend comic con. We went to the schedules section of team to try and work this out. While doing this it occurred to me that this could make a coding challenge. However, I haven't been able to think it through yet so I thought I would share with you all.

Here are the points I have so far:

Coding challenge: When am I working next:
Number of shift types: 2
Shift types: Day and Night
Duration: 12 hours
Working days: 4
Off days: 4 or 3 if going from day to night
shift type cycle: 1 month (so every month one goes from day to night and vice versa)
user input: last known day of shift, (is it) night or day
return: next day, weeks, month's schedule

Q: Can you spot any obvious questions I haven't accounted for yet?
Q: Can you foresee any blockers to solving this with code; i.e. Is this a realistic project?
Q: Any other thoughts?
Q: Final one (and biased) any Pythonistas want to tackle this together?

**Comment by eddiejaoude** [comment]
Sorry just seeing this Mikey! I am still catching up on notifications 😱 Was there any progress with this?

I think it is possible, it will need certain rules, a rules engine could help, I googled and found this one https://github.com/ssh-mitm/business-rule-engine
